gtsfm/frontend/detector_descriptor/superpoint.py

```diff
@@ -0,0 +1,78 @@
+"""Superpoint detector+descriptor implementation.
+
+The network was proposed in 'SuperPoint: Self-Supervised Interest Point Detection and Description' and is implemented
+by wrapping over the authors' implementation.
+
+References:
+- https://arxiv.org/abs/1712.07629
+- https://github.com/magicleap/SuperGluePretrainedNetwork
+
+Authors: Ayush Baid
+"""
+from pathlib import Path
+from typing import Tuple, Union
+
+import numpy as np
+import torch
+
+import gtsfm.utils.images as image_utils
+from gtsfm.common.image import Image
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.detector_descriptor.detector_descriptor_base import DetectorDescriptorBase
+from thirdparty.SuperGluePretrainedNetwork.models.superpoint import SuperPoint
+
+ROOT_PATH = Path(__file__).resolve().parent.parent.parent.parent
+MODEL_WEIGHTS_PATH = (
+    ROOT_PATH / "thirdparty" / "SuperGluePretrainedNetwork" / "models" / "weights" / "superpoint_v1.pth"
+)
+
+
+class SuperPointDetectorDescriptor(DetectorDescriptorBase):
+    """Superpoint Detector+Descriptor implementation."""
+
+    def __init__(self, use_cuda: bool = True, weights_path: Union[Path,str] = MODEL_WEIGHTS_PATH) -> None:
+        """Configures the object.
+
+        Args:
+            max_keypoints: max keypoints to detect in an image.
+            is_cuda (optional): flag controlling the use of GPUs via CUDA. Defaults to True.
```

Comment:
the function has argument `use_cuda` and the docstring has `is_cuda`.

Reply:
thanks -- fixed